sdk/core/Azure.Core.Experimental/tests/DynamicJsonTests.cs

```diff
@@ -405,6 +405,297 @@ public void DisposingAChildDisposesTheParent()
             Assert.Throws<ObjectDisposedException>(() => { var foo = json.Foo; });
         }
 
+        [Test]
+        public void CanGetCamelCasePropertyEitherCase()
+        {
+            string json = """{ "foo" : 1 }""";
+
+            dynamic dynamicJson = new BinaryData(json).ToDynamic();
+
+            Assert.AreEqual(1, (int)dynamicJson.foo);
+            Assert.AreEqual(1, (int)dynamicJson.Foo);
```

Comment:
Shouldn't this fail? I thought you would need to set PascalCaseGetters here.

Reply:
How would users be expected to know this? The default is currently called `None`.

Reply:
Ah I think I misunderstood - if no mapping is specified, then the behavior is PascalCaseGettersCamelCaseSetters.

Reply:
I have to say it is slightly unintuitive that the default value for the mapping changes the default behavior that is used when not passing anything.

Reply:
My point is just that the default behavior when the enum is not specified doesn't match the enum default of None. I believe this is unusual in our APIs. I like your suggestion of moving PascalCaseGettersCamelCaseSetters to the 0 value, but in that case it would be awkward to have None as something other than 0, so perhaps that would need to be renamed to something like RawJson.

Reply:
I suppose it is less about what is in the 0th position and more about the fact that `None` changes the default behavior. Another idea would be to rename `None` to something more descriptive.

Reply:
That makes sense.

We're trying to mirror the STJ APIs as much as possible, but there isn't an exact mapping for this one given the different requirements of this type.  The closest other types are things like:
- https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.propertynamingpolicy
- https://learn.microsoft.com/en-us/dotnet/api/system.text.json.nodes.jsonnodeoptions.propertynamecaseinsensitive
- https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonnumberhandling

Interestingly, there is this one:
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializerdefaults

That lets you change the default settings of the JsonSerializer.  If we felt strongly enough, we could add this to client options with values like `General` and `Azure`, where the default was `Azure` for libraries unless they changed it.  Although we would need to find a way to plumb that through to the ToDynamic() call, which is currently independent of any library details.

@KrzysztofCwalina, do you have any strong feelings about the defaults here?  @JoshLove-msft is right that most enums are structured with `None` or `Strict` in the zeroth position and this typically maps to the default.  Is there a BCL precedent for doing otherwise?

Reply:
Enum value 0 should correspond to the default, i.e. no passing in enum argument should be equivalent to passing the 0 value. But I think this is exactly what these APIs do today, don't they? i.e. if I call response.ToDynamic() and response.ToDynamic((JsonMappingOptions)0) I will get the exact same behavior, won't i?

Reply:
No, I had configured the default to be `PascalCaseGettersCamelCaseSetters` because Azure REST API guidance is to camelCase JSON, and .NET guidance is to PascalCase properties, and given that it seems like, for Azure, we would want that behavior in the most common case, rather than asking callers to call `response.Content.ToDynamic(DynamicJsonNameMapping.PascalCaseGettersCamelCaseSetters)` everywhere.  But I can change that if we prefer that approach - it just looks rather awkward to me.

Reply:
After considering it overnight, I do think that if the default doesn't pass through the casing, that would be surprising behavior for a user, e.g. I do:

```
var json = response.Content.ToDynamic();
json.Value = "PascalCaseValue";
```

and then ToString prints:

```
{
  "value" : "PascalCaseValue";
}
```

I still don't love the 57 chars that `DynamicJsonNameMapping.PascalCaseGettersCamelCaseSetters` requires, though, so I'm going to look at adding a `DynamicJsonOptions.AzureDefault` (a savings of 25 chars!), to see if that feels a little better.